### ISSUE TITLE
no-unmodified-loop-condition throwing for objects that mutate and condition using comparison operators

### ISSUE THREAD

**Issue by danielepiccone** [eslint/eslint#10856]
**Tell us about your environment**

* **ESLint Version:**
v5.0.1

* **Node Version:**
v8.11.3

* **npm Version:**
5.6.0

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
const config = {
  extends: ["pretty-standard"],
  plugins: ["import", "prettier"],
  rules: {
    "prettier/prettier": "error",
    "import/no-unresolved": ["error", { commonjs: true }],
    "import/no-extraneous-dependencies": [
      "error",
      {
        // devDependencies: ["lib/__tests__/**/*.js"],
        optionalDependencies: false,
        peerDependencies: false
      }
    ]
  }
};
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

I have a case of an object, which returns its value through overriding `.valueOf()` in its prototype. This object is used within a while loop with a condition check that uses comparison operators. 

When running eslint on this block I get an error from the `no-unmodified-loop-condition` rule
```
while (pointer < to) {
   // do something
   pointer.add();
}
```

While running eslint on this other block, which uses a getter to access the object value, does not trigger any value
```
while (pointer.get() < to) {
   // do something
   pointer.add();
}

// also working with

while (pointer.valueOf() < to) {
   // do something
   pointer.add();
}
```

I am not sure what should be expected in this case, it is debatable if the object should override .valueOf() in the first place, but I guess it is reasonable to think that the linter rule should work with or without the explicit call under the assumption that every method on the object could cause a mutation.

I have added a test case for this at https://github.com/danielepiccone/eslint/commit/7954993347ee6c9711343e51becc355cdc000b3b

full code:

```js
let pointer = {
    _value: 0,

    add() {
        this._value++;
    },

    valueOf() {
        return this._value;
    }
};

let to = 5;

while (pointer < to) {
    pointer.add();
    console.log("pointer value", pointer._value);
}
```

```bash
npx eslint test.js
```

**What did you expect to happen?**
No error to get thrown

**What actually happened? Please include the actual, raw output from ESLint.**
/test/test.js
  15:8   error  'pointer' is not modified in this loop  no-unmodified-loop-condition
  15:18  error  'to' is not modified in this loop       no-unmodified-loop-condition

✖ 2 problems (2 errors, 0 warnings)

**Comment by not-an-aardvark** [comment]
Thanks for the report. This is working as intended, but it's possible we could add an option to the rule to consider any method call on an object to be mutating it. Let's see if anyone else on the team would be in favor of doing that.

Note that this was also suggested in https://github.com/eslint/eslint/issues/8021.
